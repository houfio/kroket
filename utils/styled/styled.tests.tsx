import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import * as React from 'react';

import { useStyled } from '.';

it('should add the class name', () => {
  const { result: { current: Component } } = renderHook(() => useStyled('div')`
    background-color: yellow;
  `);

  const { asFragment } = render(
    <Component/>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should forward props', () => {
  const { result: { current: Component } } = renderHook(() => useStyled('div')``);

  const { asFragment } = render(
    <Component id="test">
      test
    </Component>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should append the defined classname', () => {
  const { result: { current: Component } } = renderHook(() => useStyled('div')``);

  const { asFragment } = render(
    <Component className="test"/>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should replace dynamic variables with colors', () => {
  const { result: { current: Component } } = renderHook(() => useStyled('div')`
    background-color: ${'primary'};
  `);

  const { asFragment } = render(
    <Component/>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should call variable functions', () => {
  const { result: { current: Component } } = renderHook(() => useStyled('div')`
    background-color: ${() => 'yellow'};
  `);

  const { asFragment } = render(
    <Component/>
  );

  expect(asFragment()).toMatchSnapshot();
});
