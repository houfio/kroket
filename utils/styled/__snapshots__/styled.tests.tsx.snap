// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add the class name 1`] = `
<DocumentFragment>
  <div
    class="kroket-l1kf3y"
  />
</DocumentFragment>
`;

exports[`should append the defined classname 1`] = `
<DocumentFragment>
  <div
    class="kroket-45h test"
  />
</DocumentFragment>
`;

exports[`should call variable functions 1`] = `
<DocumentFragment>
  <div
    class="kroket-l1kf3y"
  />
</DocumentFragment>
`;

exports[`should forward props 1`] = `
<DocumentFragment>
  <div
    class="kroket-45h"
    id="test"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`should replace dynamic variables with colors 1`] = `
<DocumentFragment>
  <div
    class="kroket-1uk7d0"
  />
</DocumentFragment>
`;
